src/settings/components/ExternalDataSourcesConfig/ExternalDataSourcesSettings/ExternalDataSourcesSettings.js

```diff
@@ -0,0 +1,87 @@
+import { useState } from 'react';
+import PropTypes from 'prop-types';
+import { FormattedMessage } from 'react-intl';
+import arrayMutators from 'final-form-arrays';
+
+import ExternalDataSourcesView from '../ExternalDataSourcesView/ExternalDataSourcesView';
+import ExternalDataSourcesLookup from '../ExternalDataSourcesLookup/ExternalDataSourcesLookup';
+import ExternalDataSourcesFormModal from '../ExternaldataSourcesFormModal/ExternalDataSourcesFormModal';
+import ExternalDataSourceForm from '../ExternalDataSourceForm/ExternalDataSourceForm';
+
+const CREATING = 'create';
+const VIEWING = 'view';
+
+const ExternalDataSourcesSettings = ({
+  onDelete,
+  onCancel,
+  onSave,
+  externalKbs,
+  onSubmit,
+  initialValues
+}) => {
+  const [mode, setMode] = useState(VIEWING);
+  const [externalDataSource, setExternalDataSource] = useState();
+
+  return (
+    <>
+      <ExternalDataSourcesLookup
+        externalKbs={externalKbs}
+        onCreateClick={() => setMode(CREATING)}
+        onSelectedExternalDataSource={(_e, eds) => {
+          setExternalDataSource(eds);
+        }}
+      />
+      {
+        externalDataSource &&
+          <ExternalDataSourcesView
+            externalDataSourceId={externalDataSource.id}
+            externalKbs={externalKbs}
+            initialValues={initialValues}
+            onClose={() => setExternalDataSource()}
+            onDelete={onDelete}
+            onEditCancel={onCancel}
+            onSave={onSave}
+            onSubmit={onSubmit}
+          />
+      }
+      <ExternalDataSourcesFormModal
+        initialValues={mode === CREATING ?
+          {
+            active: false,
+            activationEnabled: false,
+            rectype: 1,
+            supportsHarvesting: true,
+            type: 'org.olf.kb.adapters.GOKbOAIAdapter',
+          }
+          :
+          { ...externalDataSource }}
+        modalProps={{
+          dismissible: true,
+          label:<FormattedMessage id="ui-local-kb-admin.settings.externalDataSources.newExternalDataSource" />,
+          onClose: () => setMode(VIEWING),
```

Comment:
Instead of 3 modes, CREATE/EDIT/VIEW, we'll only need "is create form open" yes and no here.